code/messaging/src/androidTest/java/com/adobe/marketing/mobile/messaging/E2EFunctionalTests.java

```diff
@@ -1,132 +1,159 @@
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.EventType.PERSONALIZATION_REQUEST;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.Key.PERSONALIZATION;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.Key.QUERY;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.Key.SURFACES;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.PushNotificationDetailsDataKeys.DATA;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.XDMDataKeys.EVENT_TYPE;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.XDMDataKeys.REQUEST;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.XDMDataKeys.SEND_COMPLETION;
+import static com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.Messaging.XDMDataKeys.XDM;
+import static com.adobe.marketing.mobile.util.TestHelper.getDispatchedEventsWith;
+import static com.adobe.marketing.mobile.util.TestHelper.resetTestExpectations;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import com.adobe.marketing.mobile.AdobeCallback;
+import com.adobe.marketing.mobile.Edge;
+import com.adobe.marketing.mobile.Event;
+import com.adobe.marketing.mobile.EventSource;
+import com.adobe.marketing.mobile.Messaging;
+import com.adobe.marketing.mobile.MobileCore;
+import com.adobe.marketing.mobile.SDKHelper;
+import com.adobe.marketing.mobile.edge.identity.Identity;
+import com.adobe.marketing.mobile.util.TestHelper;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+public class E2EFunctionalTests {
+    static {
+        BuildConfig.IS_E2E_TEST.set(true);
+        BuildConfig.IS_FUNCTIONAL_TEST.set(false);
+    }
+
+    @Rule
+    public RuleChain rule = RuleChain.outerRule(new TestHelper.SetupCoreRule())
+            .around(new TestHelper.RegisterMonitorExtensionRule());
+
+    // --------------------------------------------------------------------------------------------
+    // Setup and teardown
+    // --------------------------------------------------------------------------------------------
+    @Before
+    public void setup() throws Exception {
+        MessagingTestUtils.cleanCache();
+        MessagingTestUtils.setEdgeIdentityPersistence(MessagingTestUtils.createIdentityMap("ECID", "80195814545200720557089495418993853789"), TestHelper.getDefaultApplication());
```

Comment:
This does not replace the ECID but creates a new entry in the identity map at index 2. Since we always use ECID value at index 1 in identity map it will not work.
We can try using `FakeIdentity` extension like Edge does [here]([https://github.com/adobe/aepsdk-edge-android/blob/1718d25ed43a00c3f25218a23602607b[…]va/com/adobe/marketing/mobile/IdentityStateFunctionalTests.java](https://github.com/adobe/aepsdk-edge-android/blob/1718d25ed43a00c3f25218a23602607b74aec0e9/code/edge/src/androidTest/java/com/adobe/marketing/mobile/IdentityStateFunctionalTests.java#L159))